### PR TITLE
Complete 5.3.15 requirement for CIS RHEL7

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

Include `rhel7` in prodtype for `sshd_use_strong_kex` rule and updated the CIS control file for RHEL7.

#### Rationale:

Better CIS coverage for RHEL7.

### CODE REVIEW

File: controls/cis_rhel7.yml
Comment (jan-cerny) [inline controls/cis_rhel7.yml:1854]: I think that you should set the variable sshd_strong_kex to some specific value.

**Review by jan-cerny — CHANGES_REQUESTED** [review]
tests on rhel7 look fine:

```
[jcerny@thinkpad scap-security-guide{pr/10238}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel7 sshd_use_strong_kex
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-02-21-1107/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_sshd_use_strong_kex
INFO - Script good_kex.pass.sh using profile (all) OK
INFO - Script no_kex.fail.sh using profile (all) OK
[jcerny@thinkpad scap-security-guide{pr/10238}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel7 --remediate-using ansible sshd_use_strong_kex
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-02-21-1109/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_sshd_use_strong_kex
INFO - Script good_kex.pass.sh using profile (all) OK
INFO - Script no_kex.fail.sh using profile (all) OK
```

File: controls/cis_rhel7.yml
Comment (marcusburghardt) [reply controls/cis_rhel7.yml:1854]: Good catch. It is now updated in the last commit. Please, take a look in the commit message for more details.

**Comment by marcusburghardt** [comment]
I have to check the error reported by `testing-farm:centos-7-x86_64`. It seems legit.

**Comment by jan-cerny** [comment]
@marcusburghardt the CI fail in centos 7 in ansible CIS hardening looks relevant to this PR:

```
:: [ 12:55:05 ] :: [   FAIL   ] :: Rules not passing after remediation:

xccdf_org.ssgproject.content_rule_sshd_use_strong_kex - fail
```

Can you take a look?

**Comment by jan-cerny** [comment]
You are 10 seconds faster than me

**Comment by jan-cerny** [comment]
@marcusburghardt  We're sorry but we had an issue with the CI jobs and unfortunately they're required to pass before merging. Could you please rebase this PR on the top of the latest upstream master branch? That will include fix from https://github.com/ComplianceAsCode/content/pull/10250 to this PR and will unblock the CI.

**Comment by marcusburghardt** [comment]
I am not sure why the `testing-farm:centos-7-x86_64` is failing. RHEL7 is passing in my local VMs.

**Review by jan-cerny — APPROVED** [review]
```
[jcerny@thinkpad scap-security-guide{pr/10238}]$ python3 tests/automatus.py rule --libvirt qemu:///system ssgts_rhel7 sshd_use_strong_kex
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-03-29-1418/test_suite.log
`WARNING - Script good_kex_sle.pass.sh is not applicable on given platform
INFO - xccdf_org.ssgproject.content_rule_sshd_use_strong_kex
INFO - Script bad_kex_rhel7_cis.fail.sh using profile xccdf_org.ssgproject.content_profile_cis OK
INFO - Script good_kex_rhel7_cis.pass.sh using profile xccdf_org.ssgproject.content_profile_cis OK
INFO - Script good_kex_strict_rhel7_cis.pass.sh using profile xccdf_org.ssgproject.content_profile_cis OK
INFO - Script mixed_kex_rhel7_cis.fail.sh using profile xccdf_org.ssgproject.content_profile_cis OK
INFO - Script no_kex.fail.sh using profile (all) OK
[jcerny@thinkpad scap-security-guide{pr/10238}]$ python3 tests/automatus.py rule --remediate-using ansible --libvirt qemu:///system ssgts_rhel7 sshd_use_strong_kex
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/work/git/scap-security-guide/logs/rule-custom-2023-03-29-1424/test_suite.log
WARNING - Script good_kex_sle.pass.sh is not applicable on given platform
INFO - xccdf_org.ssgproject.content_rule_sshd_use_strong_kex
INFO - Script bad_kex_rhel7_cis.fail.sh using profile xccdf_org.ssgproject.content_profile_cis OK
INFO - Script good_kex_rhel7_cis.pass.sh using profile xccdf_org.ssgproject.content_profile_cis OK
INFO - Script good_kex_strict_rhel7_cis.pass.sh using profile xccdf_org.ssgproject.content_profile_cis OK
INFO - Script mixed_kex_rhel7_cis.fail.sh using profile xccdf_org.ssgproject.content_profile_cis OK
INFO - Script no_kex.fail.sh using profile (all) OK
```

**Pull request merged**
